data-prepper-plugins/event-json-codecs/src/test/java/org/opensearch/dataprepper/plugins/codec/eventjson/EventJsonInputOutputCodecTest.java

```diff
@@ -0,0 +1,132 @@
+        return new EventJsonInputCodec();
+    }
+
+    @Test
+    public void emptyTest() throws Exception {
+        ByteArrayInputStream inputStream = new ByteArrayInputStream("".getBytes());
+        inputCodec = createInputCodec();
+        inputCodec.parse(inputStream, record -> { });
+        inputStream = new ByteArrayInputStream("[]".getBytes());
+        inputCodec.parse(inputStream, record -> { });
+        inputStream = new ByteArrayInputStream("[{}]".getBytes());
+        inputCodec.parse(inputStream, record -> { });
+        inputStream = new ByteArrayInputStream("{}".getBytes());
+        inputCodec.parse(inputStream, record -> { });
+        
+    }
+
+    @Test
+    public void basicTest() throws Exception {
+        final String key = UUID.randomUUID().toString();
+        final String value = UUID.randomUUID().toString();
+        Map<String, Object> data = Map.of(key, value);
+
+        Instant startTime = Instant.now();
+        Event event = createEvent(data, startTime);
+        outputCodec = createOutputCodec();
+        inputCodec = createInputCodec();
+        outputCodec.start(outputStream, null, null);
+        outputCodec.writeEvent(event, outputStream);
+        outputCodec.complete(outputStream);
+        inputCodec.parse(new ByteArrayInputStream(outputStream.toByteArray()), record -> {
+            Event e = record.getData();
+            assertThat(e.get(key, String.class), equalTo(value));
+            assertThat(e.getMetadata().getTimeReceived(), equalTo(startTime));
+            assertThat(e.getMetadata().getTags().size(), equalTo(0));
+            assertThat(e.getMetadata().getExternalOriginationTime(), equalTo(null));
+        });
+    }
+
+    @Test
+    public void extendedTest() throws Exception {
+        final String key = UUID.randomUUID().toString();
+        final String value = UUID.randomUUID().toString();
+        final String attrKey = UUID.randomUUID().toString();
+        final String attrValue = UUID.randomUUID().toString();
+        Map<String, Object> data = Map.of(key, value);
+
+        Set<String> tags = Set.of(UUID.randomUUID().toString(), UUID.randomUUID().toString());
+        List<String> tagsList = tags.stream().collect(Collectors.toList());
+        Instant startTime = Instant.now();
+        Event event = createEvent(data, startTime);
+        Instant origTime = startTime.minusSeconds(5);
+        event.getMetadata().setExternalOriginationTime(origTime);
+        event.getMetadata().addTags(tagsList);
+        event.getMetadata().setAttribute(attrKey, attrValue);
+        outputCodec = createOutputCodec();
+        inputCodec = createInputCodec();
+        outputCodec.start(outputStream, null, null);
+        outputCodec.writeEvent(event, outputStream);
+        outputCodec.complete(outputStream);
+        assertThat(outputCodec.getExtension(), equalTo(EventJsonOutputCodec.EVENT_JSON));
+        inputCodec.parse(new ByteArrayInputStream(outputStream.toByteArray()), record -> {
```

Comment:
There is no guarantee that this consumer is even called.

A simple change is to update the consumer to put the records into a collection. Then assert the size. Then assert the format.

```
List<Record> records = new LinkedList<>();
inputCodec.parse(new ByteArrayInputStream(outputStream.toByteArray()), records::add);

assertThat(records.size(), equalTo(1));
for(Record record : records) {
  Event e = record.getData();
  // All the same from here.
}
```